qiskit/algorithms/amplitude_estimators/estimation_problem.py

```diff
@@ -0,0 +1,256 @@
+        """Apply post processing to the input value.
+
+        Returns:
+            A handle to the post processing function. Acts as identity by default.
+        """
+        if self._post_processing is None:
+            return lambda x: x
+
+        return self._post_processing
+
+    @post_processing.setter
+    def post_processing(self, post_processing: Optional[Callable[[float], float]]) -> None:
+        """Set the post processing function.
+
+        Args:
+            post_processing: A handle to the post processing function.
+        """
+        self._post_processing = post_processing
+
+    @property
+    def is_good_state(self) -> Callable[[str], bool]:
+        """Checks whether a bitstring represents a good state.
+
+        Returns:
+            Handle to the ``is_good_state`` callable.
+        """
+        if self._is_good_state is None:
+            return lambda x: all(bit == '1' for bit in x)
+
+        return self._is_good_state
+
+    @is_good_state.setter
+    def is_good_state(self, is_good_state: Optional[Callable[[str], bool]]) -> None:
+        """Set the ``is_good_state`` function.
+
+        Args:
+            is_good_state: A function to determine whether a bitstring represents a good state.
+        """
+        self._is_good_state = is_good_state
+
+    @property
+    def grover_operator(self) -> Optional[QuantumCircuit]:
+        r"""Get the :math:`\mathcal{Q}` operator, or Grover operator.
+
+        If the Grover operator is not set, we try to build it from the :math:`\mathcal{A}` operator
+        and `objective_qubits`. This only works if `objective_qubits` is a list of integers.
+
+        Returns:
+            The Grover operator, or None if neither the Grover operator nor the
+            :math:`\mathcal{A}` operator is  set.
+        """
+        if self._grover_operator is not None:
+            return self._grover_operator
+
+        if self.state_preparation is not None and isinstance(self.objective_qubits, list):
+            # build the reflection about the bad state: a MCZ with open controls (thus X gates
+            # around the controls) and X gates around the target to change from a phaseflip on
+            # |1> to a phaseflip on |0>
+            num_state_qubits = self.state_preparation.num_qubits \
+                - self.state_preparation.num_ancillas
+
+            oracle = QuantumCircuit(num_state_qubits)
+            oracle.h(self.objective_qubits[-1])
+            if len(self.objective_qubits) == 1:
+                oracle.x(self.objective_qubits[0])
+            else:
+                oracle.mcx(self.objective_qubits[:-1], self.objective_qubits[-1])
+            oracle.h(self.objective_qubits[-1])
+
+            # construct the grover operator
+            return GroverOperator(oracle, self.state_preparation)
+
+        return None
+
+    @grover_operator.setter
+    def grover_operator(self, grover_operator: QuantumCircuit) -> None:
```

Comment:
This also should marked be Optional I think to allow it to be 'official' that it can be reset to the built-in default. Maybe it would be good in the setter docstrings here to note that None sets back to default and refer to constructor for more information around the default

Reply:
Yes that's a good idea, added